nil pointer in api

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xf68cdd]

goroutine 4892 [running]:
github.com/iotexproject/iotex-core/api.(*LogFilter).match(0xc0000238e8, 0xc0002a8750, 0xc00014c160)
	/go/apps/iotex-core/api/logfilter.go:82 +0x3d
github.com/iotexproject/iotex-core/api.(*LogFilter).MatchLogs(0xc0000238e8, 0xc000010fc0, 0x1, 0x1, 0x1, 0x0, 0x0)
	/go/apps/iotex-core/api/logfilter.go:72 +0xc2
github.com/iotexproject/iotex-core/api.(*Server).getLogsInBlock(0xc00002f800, 0xc0000238e8, 0x4e2a, 0x1, 0x9f9f2e2c336ae474, 0x1aa4778becb4743, 0x0, 0x0, 0x3)
	/go/apps/iotex-core/api/api.go:1359 +0x220
github.com/iotexproject/iotex-core/api.(*Server).GetLogs(0xc00002f800, 0x16be9a0, 0xc00120e7e0, 0xc0013fa0c0, 0xc00002f800, 0x7f28b02b1e00, 0xc0002e17c0)
```

--- a/api/api.go
+++ b/api/api.go
@@ -650,6 +650,9 @@ func (api *Server) GetLogs(
 	ctx context.Context,
 	in *iotexapi.GetLogsRequest,
 ) (*iotexapi.GetLogsResponse, error) {
+	if in.GetFilter() == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty filter")
+	}
 	switch {
 	case in.GetByBlock() != nil:
 		req := in.GetByBlock()
@@ -657,7 +660,7 @@ func (api *Server) GetLogs(
 		if err != nil {
 			return nil, status.Error(codes.InvalidArgument, "invalid block hash")
 		}
-		filter, ok := NewLogFilter(in.Filter, nil, nil).(*LogFilter)
+		filter, ok := NewLogFilter(in.GetFilter(), nil, nil).(*LogFilter)
 		if !ok {
 			return nil, status.Error(codes.Internal, "cannot convert to *LogFilter")
 		}
@@ -668,7 +671,7 @@ func (api *Server) GetLogs(
 		if req.FromBlock > api.bc.TipHeight() {
 			return nil, status.Error(codes.InvalidArgument, "start block > tip height")
 		}
-		filter, ok := NewLogFilter(in.Filter, nil, nil).(*LogFilter)
+		filter, ok := NewLogFilter(in.GetFilter(), nil, nil).(*LogFilter)
 		if !ok {
 			return nil, status.Error(codes.Internal, "cannot convert to *LogFilter")
 		}
@@ -699,9 +702,12 @@ func (api *Server) StreamBlocks(in *iotexapi.StreamBlocksRequest, stream iotexap
 
 // StreamLogs streams logs that match the filter condition
 func (api *Server) StreamLogs(in *iotexapi.StreamLogsRequest, stream iotexapi.APIService_StreamLogsServer) error {
+	if in.GetFilter() == nil {
+		return status.Error(codes.InvalidArgument, "empty filter")
+	}
 	errChan := make(chan error)
 	// register the log filter so it will match logs in new blocks
-	if err := api.chainListener.AddResponder(NewLogFilter(in.Filter, stream, errChan)); err != nil {
+	if err := api.chainListener.AddResponder(NewLogFilter(in.GetFilter(), stream, errChan)); err != nil {
 		return status.Error(codes.Internal, err.Error())
 	}
 

